engine-rest/engine-rest/src/test/java/org/camunda/bpm/engine/rest/PersistenceConnectionTest.java

```diff
@@ -0,0 +1,151 @@
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.camunda.bpm.engine.rest;
+
+import static io.restassured.RestAssured.given;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.camunda.bpm.engine.rest.PersistenceConnectionTest.UserQueryFactory.createFailingQueryStub;
+import static org.camunda.bpm.engine.rest.exception.ExceptionLogger.REST_API;
+import static org.camunda.bpm.engine.rest.helper.MockProvider.EXAMPLE_USER_FIRST_NAME;
+import static org.hamcrest.Matchers.equalTo;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.ibatis.exceptions.PersistenceException;
+import org.camunda.bpm.engine.ProcessEnginePersistenceException;
+import org.camunda.bpm.engine.identity.UserQuery;
+import org.camunda.bpm.engine.rest.util.container.TestContainerRule;
+import org.camunda.commons.testing.ProcessEngineLoggingRule;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+public class PersistenceConnectionTest extends AbstractRestServiceTest {
+
+  @ClassRule
+  public static TestContainerRule rule = new TestContainerRule();
+
+  @Rule
+  public ProcessEngineLoggingRule loggingRule = new ProcessEngineLoggingRule()
+      .watch(REST_API);
+
+  protected static final String USER_QUERY_URL = TEST_RESOURCE_ROOT_PATH + "/user";
+
+  private final ConnectionSubclass subclass;
+
+  public PersistenceConnectionTest(ConnectionSubclass subclass) {
+    this.subclass = subclass;
+  }
+
+  @Parameters(name = "{index}: {0}")
+  public static Collection<Object[]> data() {
+    ConnectionSubclass[] values = ConnectionSubclass.values();
+
+    return Arrays.stream(values)
+        .map(c -> new Object[] {c})
+        .collect(Collectors.toList());
+  }
+
+  @Test
+  public void shouldLogError() {
+    createFailingQueryStub(subclass);
+
+    String expectedMessage = "This is an exception originating from the engine persistence layer";
```

Comment:
❓ Is there a reason we are not using the same error message here as in the exceptions created through [ExceptionUtil.wrapPersistenceException()](https://github.com/camunda/camunda-bpm-platform/blob/f8625f0839be73c7d66633e5329b1d635c3f180e/engine/src/main/java/org/camunda/bpm/engine/impl/util/ExceptionUtil.java#L356)? We could reference the [message directly from that class](https://github.com/camunda/camunda-bpm-platform/blob/f8625f0839be73c7d66633e5329b1d635c3f180e/engine/src/main/java/org/camunda/bpm/engine/impl/util/ExceptionUtil.java#L45) to make the test more robust.

Reply:
[ExceptionUtil.wrapPersistenceException()](https://github.com/camunda/camunda-bpm-platform/blob/f8625f0839be73c7d66633e5329b1d635c3f180e/engine/src/main/java/org/camunda/bpm/engine/impl/util/ExceptionUtil.java#L356) is an implementation detail. The test is written in an agnostic way when it comes to the specific message of the exception, emphasising to test if the exception with be logged with ERROR if its a persistence connection exception.

❓ According to your point of view, how would the test become more robust if we use the implementation to create the exception ?